source/common/http/conn_pool_grid.cc

```diff
@@ -356,8 +351,7 @@ void ConnectivityGrid::addIdleCallback(IdleCb cb) {
 
 void ConnectivityGrid::drainConnections(Envoy::ConnectionPool::DrainBehavior drain_behavior) {
   if (drain_behavior == Envoy::ConnectionPool::DrainBehavior::DrainAndDelete) {
-    // Note that no new pools can be created from this point on
-    // as createNextPool fast-fails if `draining_` is true.
+    // Note that no new pools should be created from this point on.
```

Comment:
nit: "should be" => "will be" since this invariant is enforced.

Reply:
I don't think it is, there's just an assert?
we  used to handle null pools, now we'll create on drain.